internal/indexer/processors/contract_deploy.go

```diff
@@ -0,0 +1,110 @@
+package processors
+
+import (
+	"context"
+	"fmt"
+
+	operation_processor "github.com/stellar/go/processors/operation"
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/wallet-backend/internal/indexer/types"
+)
+
+// ContractDeployProcessor emits state changes for contract deployments.
+type ContractDeployProcessor struct {
+	networkPassphrase string
+}
+
+func NewContractDeployProcessor(networkPassphrase string) *ContractDeployProcessor {
+	return &ContractDeployProcessor{networkPassphrase: networkPassphrase}
+}
+
+// ProcessOperation emits a state change for each contract deployment (including subinvocations).
+func (p *ContractDeployProcessor) ProcessOperation(_ context.Context, op *operation_processor.TransactionOperationWrapper) ([]types.StateChange, error) {
+	if op.OperationType() != xdr.OperationTypeInvokeHostFunction {
+		return nil, ErrInvalidOpType
+	}
+	invokeHostOp := op.Operation.Body.MustInvokeHostFunctionOp()
+
+	opID := op.ID()
+	builder := NewStateChangeBuilder(op.Transaction.Ledger.LedgerSequence(), op.LedgerClosed.Unix(), op.Transaction.Hash.HexString()).
+		WithOperationID(opID).
+		WithCategory(types.StateChangeCategoryContract).
+		WithReason(types.StateChangeReasonDeploy)
+
+	deployedContractsMap := map[string]types.StateChange{}
+
+	processCreate := func(fromAddr xdr.ContractIdPreimageFromAddress) error {
```

Comment:
I am just curious - what is the recommended way here: do we create a private function as a variable or do we assign it to the ContractDeployProcessor struct. Normally I have always done the latter 🤔